### PR TITLE
Composition and layout refactor

### PR DESCRIPTION
**Pull request by jaredhms**
This change refactors WinObjC's QuartzCore and UIKit stacks to align more closely with Xaml, as detailed in the docs/CoreAnimation/WinObjc.Composition.and.Layout.docx dev spec.

Some highlights include:

    1.	Streamlines the usage of arbitrary Xaml FrameworkElements by allowing them to be positioned just as any other UIView in the application.
    2.	Supports rendering over Xaml-backed UIKit controls; adding adornments, etc.
    3.	Supports adding subviews and sublayers to Xaml-backed UIKit controls.
    4.	Separates the complex details of our Core Animation composition layer into distinct and manageable subcomponents  improving maintainability.
    5.	Defines a concrete relationship between every UIView, its root CALayer, and its backing Xaml FrameworkElement.
    6.	Streamlines the UIElement tree by removing unnecessary UIElements (for example; this results in a ~30% reduction of UIElements on the initial scene of WOCCatalog).
    7.	Simplifies app rendering by removing our two custom Panel implementations; thus eliminating the need to take part in Arrange and Measure passes.
    8.	Paves the way for eventual Xaml-markup-driven UIElement trees in ported WinObjC apps, because we no longer require re-parenting Xaml FrameworkElements that are positioned by Core Animation.
    9.	Moves the rest of the Core Animation composition layer over to C++, to benefit from modern programming constructs such as RAII, external ref-counting (shared_ptr), etc.

In order to implement these changes, we had to move UILabel over to a fully Xaml-backed implementation (rather than following a more complicated path with fake 'textures' down into the compositor).  As a result, there are some known issues around text layout in one of our internal test apps (MM); we'll address those issues (and anything else that comes up) in a stabilization pass after these changes go in.

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/winobjc/1317)

### CODE REVIEW

**Comment by jaredhms** [comment]
@oliversa-msft is added to the review. #Closed

**Comment by jaredhms** [comment]
@yiyang-msft is added to the review. #Closed

**Comment by jaredhms** [comment]
@rajsesh-msft is added to the review. #Closed

**Comment by jaredhms** [comment]
@bbowman is added to the review. #Closed

**Comment by jaredhms** [comment]
@DHowett-MSFT is added to the review. #Closed

**Comment by jaredhms** [comment]
@msft-Jeyaram is added to the review. #Closed

**Comment by jaredhms** [comment]
@davelamb is added to the review. #Closed

**Comment by jaredhms** [comment]
@ashvarma is added to the review. #Closed

**Comment by jaredhms** [comment]
@bviglietta is added to the review. #Closed

**Comment by jaredhms** [comment]
@tadam-msft is added to the review. #Closed

**Comment by jaredhms** [comment]
oops; this should have been a PR for ux/uxe.  i'll cancel and resubmit.

File: Frameworks/UIKit/UIActivityIndicatorView.mm [hunk 1]
```diff
@@ -107,12 +107,21 @@ - (instancetype)initWithActivityIndicatorStyle:(UIActivityIndicatorViewStyle)sty
...
+
+    return self;
 }
 
-- (instancetype)_initWithFrame:(CGRect)frame xamlElement:(WXFrameworkElement*)xamlElement {
-    if (self = [super initWithFrame:frame]) {
-        [self _UIActivityIndicatorView_initInternal:xamlElement];
+/**
+ Microsoft Extension
+*/
```
Comment (DHowett-MSFT) [inline Frameworks/UIKit/UIActivityIndicatorView.mm]: Is there a `@Status` we can use for this? That'd be cool.

**Pull request closed without merging**